### PR TITLE
Update request library

### LINKED ISSUE #8: IE11 CORS error fetching vimeo thumbnail image
Here's a strange one. On IE11, fetching the YouTube image works fine and returns an object `{err: null, thumbnail: {...}}`

But the exact same code requesting the thumbnail image from a vimeo video returns an error:
`SEC7118: XMLHttpRequest for http://vimeo.com/api/v2/video/200927594.json required Cross Origin Resource Sharing (CORS).`

This alternative code does work in IE11. It uses the jsonp API from Vimeo
http://jsfiddle.net/archatas/Tv7GZ/

I ended up using the npm module `fetch-jsonp` to fetch vimeo thumbs instead of embed-video. But you could patch your plugin using fetch-jsonp to get around this issue in IE11
```
import fetchJsonp from 'fetch-jsonp'
...
fetchJsonp('http://vimeo.com/api/v2/video/' + vimeoId + '.json')
.then((response)=> {
  return response.json()
}).then((json)=> {
  console.log('parsed json', json)
  let copiedState = Object.assign([],this.state.imageSources)
  copiedState[i] = json[0].thumbnail_small
  this.setState({imageSources: copiedState})
}).catch((ex)=> {
  console.log('parsing failed', ex)
})
```

### LINKED ISSUE #12: Using `request` makes this only available in Node
We should use an AJAX library that works both in node and the browser.

http://andrewhfarmer.com/ajax-libraries/

### PR DESCRIPTION
**Pull request by treyhuffine**
Closes #12 and closes #8. Use the `axios` library which works on the client and server with unchanged browser compatability.

### CODE REVIEW

**Comment by treyhuffine** [comment]
@alanshaw can we update the version of node that Travis CI is using to run tests?

File: index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 var URL = require('url');
-var request = require('request');
+var promise = require('promise-polyfill');
+var { fetch } = require('fetch-ponyfill')({ Promise: promise });
```
Comment (alanshaw) [inline index.js]: There's no transpilation in this project so please don't use destructuring assignment as it's not widely supported in browsers. Please change to:

```js
var fetch = require('fetch-ponyfill')({ Promise: promise }).fetch;
```

File: index.js [hunk 2]
```diff
@@ -185,23 +186,27 @@ embed.vimeo.image = function (id, opts, cb) {
...
-  request.get({
-    url: 'http://vimeo.com/api/v2/video/' + id + '.json',
-    json: true
-  }, function (err, res, body) {
-    if (err) return cb(err)
-    if (res.statusCode !== 200) return cb(new Error('unexpected response from vimeo'))
-    if (!body || !body[0] || !body[0][opts.image]) return cb(new Error('no image found for vimeo.com/' + id))
+  fetch('http://vimeo.com/api/v2/video/' + id + '.json')
+    .then(function (res) {
+      if (res.status !== 200) return cb(new Error('unexpected response from vimeo'))
```
Comment (alanshaw) [inline index.js]: Please throw this error so the catch block can call the callback

File: index.js [hunk 3]
```diff
@@ -185,23 +186,27 @@ embed.vimeo.image = function (id, opts, cb) {
...
 
-    var src = body[0][opts.image].split(':')[1]
+      return res.json();
+    })
+    .then(function (res) {
+      var body = res;
+      if (!body || !body[0] || !body[0][opts.image]) return cb(new Error('no image found for vimeo.com/' + id))
```
Comment (alanshaw) [inline index.js]: Again please throw

File: index.js [hunk 4]
```diff
@@ -185,23 +186,27 @@ embed.vimeo.image = function (id, opts, cb) {
...
-    if (!body || !body[0] || !body[0][opts.image]) return cb(new Error('no image found for vimeo.com/' + id))
+  fetch('http://vimeo.com/api/v2/video/' + id + '.json')
+    .then(function (res) {
+      if (res.status !== 200) return cb(new Error('unexpected response from vimeo'))
 
-    var src = body[0][opts.image].split(':')[1]
+      return res.json();
+    })
+    .then(function (res) {
+      var body = res;
```
Comment (alanshaw) [inline index.js]: You might as well call the function parameter `body` and then you don't need to do this extra assignment

File: index.js [hunk 5]
```diff
@@ -216,22 +221,27 @@ embed.dailymotion.image = function(id, opts, cb) {
...
-    var src = body[opts.image]
+  fetch('https://api.dailymotion.com/video/' + id + '?fields=' + opts.image)
+    .then(function (res) {
+      if (res.status !== 200) return cb(new Error('unexpected response from dailymotion'))
 
-    var result = {
+      return res.json();
+    })
+    .then(function(res) {
+      var body = res;
```
Comment (alanshaw) [inline index.js]: You might as well call the function parameter `body` and then you don't need to do this extra assignment

File: index.js [hunk 6]
```diff
@@ -216,22 +221,27 @@ embed.dailymotion.image = function(id, opts, cb) {
...
-    url: 'https://api.dailymotion.com/video/' + id + '?fields=' + opts.image,
-    json: true
-  }, function(err, res, body) {
-    if (err) return cb(err)
-    if (res.statusCode !== 200) return cb(new Error('unexpected response from dailymotion'))
-    if (!body || !body[opts.image]) return cb(new Error('no image found for dailymotion.com/' + id))
-    var src = body[opts.image]
+  fetch('https://api.dailymotion.com/video/' + id + '?fields=' + opts.image)
+    .then(function (res) {
+      if (res.status !== 200) return cb(new Error('unexpected response from dailymotion'))
```
Comment (alanshaw) [inline index.js]: Please throw

File: index.js [hunk 7]
```diff
@@ -216,22 +221,27 @@ embed.dailymotion.image = function(id, opts, cb) {
...
 
-    var result = {
+      return res.json();
+    })
+    .then(function(res) {
+      var body = res;
+      if (!body || !body[opts.image]) return cb(new Error('no image found for dailymotion.com/' + id))
```
Comment (alanshaw) [inline index.js]: Please throw

File: index.js [hunk 8]
```diff
@@ -185,23 +186,27 @@ embed.vimeo.image = function (id, opts, cb) {
...
+        src: src,
+        html: '<img src="' + src + '"/>'
+      }
 
-    cb(null, result)
-  })
+      cb(null, result)
+    })
+    .catch(function(err) {
+      return cb(err)
```
Comment (alanshaw) [inline index.js]: No need to return, this is the last (and only) statement

**Comment by alanshaw** [comment]
@treyhuffine also yes, feel free to update the travis node version - 4 & 6?

File: index.js (hunk 1 above)
Comment (treyhuffine) [inline index.js]: Good catch, did it out of habit

**Comment by alanshaw** [comment]
:thumbsup: thank you!

**Pull request merged**